python/lsst/pipe/base/config.py

```diff
@@ -46,249 +42,81 @@
+                                                                        doc=f"name for "
+                                                                            "connection {fieldName}",
+                                                                        default=obj.name)
+            # If there are default templates also add them as fields to configure the template values
+            if hasattr(connectionsClass, 'defaultTemplates'):
+                docString = "Template parameter used to format corresponding field template parameter"
+                for templateName, default in connectionsClass.defaultTemplates.items():
+                    configConnectionsNamespace[templateName] = pexConfig.Field(dtype=str,
+                                                                               doc=docString,
+                                                                               default=default)
+            # add a reference to the connection class used to create this sub config
+            configConnectionsNamespace['connectionsClass'] = connectionsClass
+
+            # Create a new config class with the fields defined above
+            Connections = type("Connections", (pexConfig.Config,), configConnectionsNamespace)
+            # add it to the Config class that is currently being declared
+            dct['connections'] = pexConfig.ConfigField(dtype=Connections,
+                                                       doc='Configurations describing the'
+                                                           'connections of the PipelineTask to datatypes')
+            dct['ConnectionsConfigClass'] = Connections
+        inst = super().__new__(cls, name, bases, dct)
+        return inst
+
+    def __init__(self, name, bases, dct, **kwargs):
+        # This overrides the default init to drop the kwargs argument. Python metaclasses will
+        # have this argument set if any kwargs are passes at class construction time, but should
+        # be consumed before calling __init__ on the type metaclass. This is in accordance with
+        # python documentation on metaclasses
+        super().__init__(name, bases, dct)
+
+
+class PipelineTaskConfig(pexConfig.Config, metaclass=PipelineTaskConfigMeta):
+    """Configuration class for `PipelineTask`s
+
+    This Configuration class functions in largely the same manor as any other derived from
+    `lsst.pex.config.Config`. The only difference is in how it is declared. `PipelineTaskConfig` children need
+    to be declared with a pipelineConnections argument. This argument should specify a child class of
+    `PipelineTaskConnections`. During the declaration of a `PipelineTaskConfig` a config class is created with
+    information from the supplied connections class to allow configuration of the connections class. This
+    dynamically created config class is then attached to the `PipelineTaskConfig` via a
+    `lsst.pex.config.ConfigField` with the attribute name connections.
```

Comment:
So in this case, config should not be a problem with pickle because it implements its own reduce functionality (and it is not a problem as it is currently used successfully inside a python multiprocessing environment).